src/components/NotFound/NotFound.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import styles from './NotFound.module.css';
+import { useNavigate } from 'react-router';
+import useComponentTransition from '../../utils/custom-hooks/useComponentTransition';
+import { animated } from '@react-spring/web';
+
+const NotFound = () => {
+  const [progress, setProgress] = React.useState(false);
+  const navigate = useNavigate();
+  const { transition, setShowComponent } = useComponentTransition(
+    { y: -500, opacity: 0.5 },
+    { y: 0, opacity: 1 },
+    { y: -300, opacity: 0.5}
+  );
+
+  React.useEffect(() => {
```

Comment:
This is a bad practice. You should not redirect users from your 404 page. It might create unwanted behaviours where users are sharing your 404 pages by mistake (because they know they are going to redirect them anyway).